processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go

```diff
@@ -74,49 +70,62 @@ func newTraceProcessor(nextConsumer consumer.TraceConsumer, cfg Config) (compone
 }
 
 func (tsp *tracesamplerprocessor) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
-	octds := internaldata.TraceDataToOC(td)
 	var errs []error
-	for _, octd := range octds {
-		if err := tsp.processTraces(ctx, octd); err != nil {
+	rspans := td.ResourceSpans()
+	sampledTraceData := pdata.NewTraces()
+	for i := 0; i < rspans.Len(); i++ {
+		rspan := rspans.At(i)
+		if rspan.IsNil() {
+			continue
+		}
+		if err := tsp.processTraces(ctx, rspan, sampledTraceData); err != nil {
```

Comment:
It doesn't return error anymore. so we can stop accumulating them

Reply:
yes, removed it. Thanks